src/ui/ui_settings/ui_settings_service_for_request.js

```diff
@@ -0,0 +1,20 @@
+import { uiSettingsServiceFactory } from './ui_settings_service_factory';
+
+const BY_REQUEST_CACHE = new WeakMap();
+
+export function getUiSettingsServiceForRequest(server, request, readInterceptor) {
+  if (BY_REQUEST_CACHE.has(request)) {
+    return BY_REQUEST_CACHE.get(request);
+  }
+
+  const adminCluster = server.plugins.elasticsearch.getCluster('admin');
+  const uiSettingsServices = uiSettingsServiceFactory(server, {
+    readInterceptor,
+    callCluster(...args) {
```

Comment:
I chose not to include `admin` in the name because it's really not a concern of the `UiSettingsService` what type of cluster it's calling. `UiSettingsService` just cares that it can talk to elasticsearch.